src/masternodes/rpc_vault.cpp

```diff
@@ -485,6 +489,10 @@ UniValue listvaults(const JSONRPCRequest& request) {
         if (!ownerAddress.empty() && ownerAddress != data.ownerAddress) {
             return false;
         }
+
+        if (isMine && !(IsMineCached(*pwallet, data.ownerAddress) & ISMINE_SPENDABLE)) {
```

Comment:
Does it have to spendable? Could you verify with @bvbfan on this?

Reply:
Yup, we don't show watch only addresses, it could be added in the future.